pyne/alara.py

```diff
@@ -298,6 +398,78 @@ def photon_source_hdf5_to_mesh(mesh, filename, tags, sub_voxel=False,
                     temp_mesh_data[i, :].reshape(max_num_cells * num_e_groups)
 
 
+def response_hdf5_to_mesh(mesh, filename, tags, response):
+    """This function reads in an hdf5 file produced by response_to_hdf5
+    and tags the requested data to the mesh of a PyNE Mesh object. Any
+    combinations of nuclides and decay times are allowed. The photon source
+    file is assumed to be in mesh.__iter__() order
+
+    Parameters
+    ----------
+    mesh : PyNE Mesh
+       The object containing the PyMOAB instance to be tagged.
+    filename : str
+        The path of the hdf5 version of the photon source file.
+    tags: dict
+        A dictionary were the keys are tuples with two values. The first is a
+        string denoting an nuclide in any form that is understood by
+        pyne.nucname (e.g. '1001', 'U-235', '242Am') or 'TOTAL' for all
+        nuclides. The second is a string denoting the decay time as it appears
+        in the file (e.g. 'shutdown', '1 h' '3 d'). The values of the
+        dictionary are the requested tag names for the combination of nuclide
+        and decay time. For example if one wanted tags for the photon source
+        densities from U235 at shutdown and from all nuclides at 1 hour, the
+        dictionary could be:
+
+        tags = {('U-235', 'shutdown') : 'tag1', ('TOTAL', '1 h') : 'tag2'}
+    response : str
+        The keyword of the response. Supported responses:
+            - decay_heat
+            - specific_activity
+            - alpha_heat
+            - beta_heat
+            - gamma_heat
+            - wdr
+            - photon_source
+    """
+
+    # create a dict of tag handles for all keys of the tags dict
+    tag_handles = {}
+    for tag_name in tags.values():
+
+        mesh.tag(tag_name, np.zeros(1, dtype=float), 'nat_mesh',
+                 size=1, dtype=float)
+        tag_handles[tag_name] = mesh.get_tag(tag_name)
+
+    phtn_src_dc = _read_phtn_src_dc(filename)
```

Comment:
This can be used for responses other than photon source, right?  Is this variable name confusing?

Reply:
It seems that there are more places below where the name based on photon source is used

Reply:
This function can be used for both photon source and the responses. For better consistency, I removed `phtn_src` and rename the `dc` to `decay_times` in https://github.com/pyne/pyne/pull/1123/commits/9a61ffa550cfa61a87cefb8c580d2a20817d8696.